src/hooks/video/agora/useAgoraScreenShare.ts

```diff
@@ -48,8 +49,10 @@ export const useAgoraScreenShare: ReactHook<
   }, [venueId, userId, client, localAudioTrack, localScreenTrack]);
 
   const leaveChannel = useCallback(async () => {
+    if (!client) return;
+
     stopShare();
-    await client?.leave();
+    await client.leave();
   }, [client, stopShare]);
```

Comment:
Is it correct to not call `stopShare` when `!client`? The previous way would have ensured the backend data stayed in sync with the expected reality I would have thought? Whereas now I expect there is more chance of them being weirdly out of sync.

Reply:
@0xdevalias at the beginning I consider in return !localScreenTrack || !localAudioTrack. But the issue was that when tapping on the button "Stop sharing" in the browser, localScreenTrack and localAudioTrack are undefined, so the streaming can't be stoped.

Reply:
By ‘the previous way’ I mean the literal code change shown in this diff here. Without the guard statement at all, and using the ? when calling .leave

Reply:
@0xdevalias so it's ok now, right?

Reply:
@0xdevalias fixed in https://github.com/sparkletown/sparkle/pull/1653/commits/1a95556a6dfecdc953e99c7e694c1f6b703180f9